apps/snitch_core/test/domain/shipping_calculator_test.exs

```diff
@@ -4,141 +4,223 @@ defmodule Snitch.Domain.ShippingCalculatorTest do
 
   import Mox
   import Snitch.Factory
-  alias Snitch.Data.Schema.Package
+  alias Snitch.Data.Schema.{Package, ShippingRule}
   alias Snitch.Domain.ShippingCalculator
 
   setup do
     Application.put_env(:snitch_core, :defaults, currency: :USD)
   end
 
-  describe "calculate/1" do
-    setup :zones
-    setup :shipping_methods
-    setup :embedded_shipping_methods
+  setup :zones
+  setup :shipping_methods
+  setup :embedded_shipping_methods
 
-    test "check for `free for order above some amount`", context do
-      expect(Snitch.Tools.DefaultsMock, :fetch, 3, fn :currency -> {:ok, :USD} end)
-      rule_active_manifest = [true, false, true, false]
-      cost_manifest = [Money.new!(:USD, 0), Money.new!(:USD, 100), Money.new!(:USD, 10)]
+  describe "fixed rate per order :ofr" do
+    test "returns only cost for :ofr", context do
+      rule_active_manifest = %{fso: false, fsoa: false, ofr: true, fsrp: false}
 
-      # set 12 items so total cost is greater than lower_limit of 100
-      package = setup_package_with_shipping(context, 12, cost_manifest, rule_active_manifest)
+      preference_manifest = %{
+        fso: %{},
+        fsoa: %{amount: 100},
+        ofr: %{cost: 20},
+        fsrp: %{cost_per_item: 5}
+      }
+
+      %{package: package, category: category} = setup_package_with_sc(context, _quantity = 5)
+
+      setup_shipping_rules(rule_active_manifest, preference_manifest, category)
+
+      shipping_cost = ShippingCalculator.calculate(package)
+
+      assert shipping_cost == Money.new!(currency(), 20) |> Money.round()
+    end
+
+    test "with free for order above amount, returns 0 as :fsoa applies", context do
+      rule_active_manifest = %{fso: false, fsoa: true, ofr: true, fsrp: false}
+
+      preference_manifest = %{
+        fso: %{},
+        fsoa: %{amount: 50},
+        ofr: %{cost: 20},
+        fsrp: %{cost_per_item: 5}
+      }
+
+      %{package: package, category: category} = setup_package_with_sc(context, _quantity = 7)
```

Comment:
make product price explicit here.

Reply:
made all the changes suggested for this file.